rfcs/rfc-2-font-measuring.md

```diff
@@ -42,13 +31,19 @@ pub struct KayakContext {
 }
 
 impl KayakContext {
-    pub fn get_asset<T>(&self, asset_handle: AssetHandle<T>) -> &T {
-        // Throw error if AssetStorage doesn't exist in the asset resources.
+    // Binding returned here to allow widgets to track loading.
+    pub fn get_asset<T>(&self, asset_handle: String) -> &Binding<Option<T>> {
+        // Create new asset if asset doesn't exist, and sets it to None.
+        ..
+    }
+ 
+    // Set's the asset value.
+    pub fn set_asset<T>(&mut self, asset_handle: String, asset: T) {
         ..
     }
 
-    // We want binding here because we need to allow widgets to track changes.
-    pub fn set_asset<T>(&mut self, asset: T) -> Binding<AssetHandle<T>> {
+    // Creates the AssetStorage<T> and stores it in assets
+    pub fn initialize_asset<T>(&mut self) {
         ..
     }
```

Comment:
Could we not handle this internally in `get_asset` or `set_asset` if the `AssetStorage` doesn't exist?

Reply:
I was thinking this would mostly be for setting it up internally.